### PR TITLE
Geant4star integration

### PR DESCRIPTION
**Pull request by klendathu2k**
Following suggestions made in PR#311...  splitting the geant4star pull request.  First committing the changes to BigFullChain.h that define the chain options needed to load in geant4 and geant4vmc support libraries.  Also adding the environment setup script and build script under StGeant4Maker/etc.  And a test macro to load the libraries.

To compile:
$ source StRoot/StGeant4Maker/etc/setup
$ source StRoot/StGeant4Maker/etc/build

To test:
$ root.exe StRoot/StGeant4Maker/macros/loadGeant4Vmc.C

Result should be a successful load of the libraries.

### CODE REVIEW

File: StRoot/StBFChain/BigFullChain.h [hunk 1]
```diff
@@ -1211,6 +1211,7 @@ Bfc_st BFC[] = { // standard chains
   {"McEvOut"     ,""  ,"","StMcEvent,Tree"                       ,"","","Write StMcEvent to StTree",kFALSE},
   {"EvOut"       ,""  ,"","Tree"                                   ,"","","Write StEvent to StTree",kFALSE},
   {"GeantOut"    ,""  ,"","Tree"                                ,"","","Write g2t tables to StTree",kFALSE},
+  {"Geant4Out"   ,""  ,"","Tree"                        ,"","","Write g2t tables to StTree from G4",kFALSE},
```
Comment (klendathu2k) [inline StRoot/StBFChain/BigFullChain.h:1214]: Note: this chain option requires an update to StBFChain.cxx to add the StGeant4Maker into the tree, which must
wait for the integration of StGeant4Maker.

File: StRoot/StBFChain/BigFullChain.h [hunk 2]
```diff
@@ -1377,12 +1378,38 @@ Bfc_st BFC[] = { // standard chains
...
   {"Generators  ","-----------","-----------","------------------------------------------","","","",kFALSE},
   {"------------","-----------","-----------","------------------------------------------","","","",kFALSE},
-  {"stargen",     "","", "gen_T,sim_T"/*+++*/,                     "", "libVMC.so,libStarGeneratorUtil.so,libStarGeneratorEvent.so,libStarGeneratorBase.so,libStarGeneratorFilt.so,libMathMore.so","STAR Generator BASE",false},
+  {"stargen.filt", "","", "gen_T,sim_T"/*+++*/,                     "", "libVMC.so,libStarGeneratorFilt.so,libMathMore.so","STAR Generator BASE",false},
+  {"stargen.util", "","", "gen_T,sim_T"/*+++*/,                     "", "libVMC.so,libStarGeneratorUtil.so,libMathMore.so","STAR Generator BASE",false},
+  {"stargen.event","","", "gen_T,sim_T"/*+++*/,                     "", "libVMC.so,libStarGeneratorEvent.so,libMathMore.so","STAR Generator BASE",false},
+  {"stargen.base", "","", "gen_T,sim_T"/*+++*/,                     "", "libVMC.so,libStarGeneratorBase.so,libMathMore.so","STAR Generator BASE",false},
+  {"stargen",     "","",  "stargen.util,stargen.event,stargen.base"/*+++*/,                     "", "","STAR Generator BASE",false},
```
Comment (klendathu2k) [inline StRoot/StBFChain/BigFullChain.h:1385]: Note: stargen loads have an unmet dependency in St_geant_Maker.  Future commits will resolve this.

**Comment by starsdong** [comment]
Jason and all, I see this one has been there for a while. As far as I see, all checks have passed, so the merge won't crash the running jobs? From your comments, Jason, there will still need other commits to resolve some of the dependencies. How should we proceed? Shall we just proceed to merge this PR (upon the sync with the latest main branch)? Thanks

File: StRoot/StGeant4Maker/etc/setup [hunk 3]
```diff
@@ -0,0 +1,25 @@
+setup 64b
+
+module use /cvmfs/star.sdcc.bnl.gov/star-spack/spack/share/spack/modules/linux-rhel7-x86_64
+#module load root-6.24.00
+source /star/simu/simu/jwebb/ROOT/local/bin/thisroot.csh
```
Comment (plexoos) [inline StRoot/StGeant4Maker/etc/setup:5]: Can you use the the ROOT installed by Spack instead? This should suffice:

```
module load star-env-X.Y.Z-root-6.V.W
```

...and you don't need to set the environment variables by hand as you do later in this file.

**Comment by klendathu2k** [comment]
As I recall, I encountered problems when trying to compile against that
environment.  Probably a missing ROOT feature... and/or needed to have a
more modern VMC version than was compiled.  These are issues I would 
prefer
to address in a future PR.

**Comment by klendathu2k** [comment]
Superseded by #617

**Pull request closed without merging**